419 fix missing exception

mandoonandy opened the pull request:
Find missing image fetch exception and add event
Fix image task bugs and add image events endpoint

Requires https://github.com/shakenfist/client-python/pull/9

Resolves #419

mikalstill commented on deploy/shakenfist_ci/base.py:
```diff
@@ -132,6 +140,33 @@ def _await_event(self, instance_uuid, operation, message=None, after=None):
+    def _await_image_download_error(self, url):
+        return self._await_image_event(
+            url, 'fetch', 'Name or service not known')
+
```
The only catch with this approach is it means failing tests always take 300 seconds to timeout, whereas if we asserted that waiting for a success means that there are no failures then we could fail faster?

mandoonandy replied:
Yes, there is a better way. I sort that out after dinner.

mikalstill reviewed (commented):
Small comment.

mikalstill commented:
CI here should now pass because the client change has been landed. We're not mature enough to be able to express CI dependancies yet.

mikalstill commented:
CI is still unhappy... https://jenkins.shakenfist.com/job/shakenfist/view/change-requests/job/PR-423/2/execution/node/19/log/

mandoonandy commented:
A few changes:

Fixed the tests to error immediately upon the wrong event.

Fixed queue handling: during image fetch exception handling, I think there was confusion of urllib HTTPError with shakenfist.exception.HTTPError

**Major change:** virt.py create() now only creates the virtual instance, it does not fetch any disk images. This removes the potential download errors from that code section. Instances are only created after a fetch so this step was redundant. (Let me know if I made a gross error?!)

Future change for Images - upon instantiation: check files exist or flag that fetch is required.

mikalstill commented:
Nice. I thought I'd already removed the fetch from virt.Instance.create(), so I am fine with that.

mikalstill reviewed (approved):
LGTM, thanks!
